Task6/AbstractFactory/PaperBookAbstractFactory.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Microsoft.Extensions.FileSystemGlobbing.Internal;
+using Microsoft.VisualBasic.FileIO;
+using System.Text.RegularExpressions;
+
+namespace Task6.AbstractFactory
+{
+    internal class PaperBookAbstractFactory : ILibrayAbstractFactory
+    {
+        List<string> _uniquePublishers;
+        int isbnNull = 1;
+
+        public Catalog<string, Book> CreateCatalog()
+        {
+            _uniquePublishers = new List<string>();
+            string filePath = "books_info.csv";
+            Catalog<string, Book> catalog = new Catalog<string, Book>();
+
+            if (!File.Exists(filePath))
+            {
+                throw new Exception("File is not in the correct path");
+            }
+
+            using (TextFieldParser parser = new TextFieldParser(filePath))
+            {
+                parser.SetDelimiters(",");
+                parser.HasFieldsEnclosedInQuotes = true;
+
+                string[] header = parser.ReadFields();
+                while (!parser.EndOfData)
+                {
+                    string[] fields = parser.ReadFields();
+                    List<Author> authorList = new List<Author>();
+                    string publicationDate = string.Empty;
+                    List<string> isbns = new List<string>();
+                    List<string> publishers = new List<string>();
+                    string title = string.Empty;
+                    bool isbnIsNull = false;
+
+                    for (int i = 0; i < header.Length; i++)
+                    {
+                        switch (header[i])
+                        {
+                            case "title":
+                                title = fields[i].Trim();
+                                break;
+                            case "publicdate":
+                                publicationDate = fields[i].Trim();
+                                break;
+                            case "creator":
+                                List<string> resultList = new List<string>();
+
```

Comment:
Move regexp patterns and objects definition  (regexp and regex2) out of the loop. They do not depend on the loop context, and no need to create objects each time to save memory

Reply:
Modified